bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceLogic.java

```diff
@@ -0,0 +1,799 @@
+		this.isTargetEditable = canEditTarget;
+		nextReplactionOperationNeedsFindOperationFirst = true;
+
+		if (this.target != newRarget) {
+			if (newRarget != null && newRarget instanceof IFindReplaceTargetExtension)
+				((IFindReplaceTargetExtension) newRarget).endSession();
+
+			this.target = newRarget;
+			if (newRarget != null)
+				fIsTargetSupportingRegEx = newRarget instanceof IFindReplaceTargetExtension3;
+
+			if (newRarget instanceof IFindReplaceTargetExtension) {
+				((IFindReplaceTargetExtension) newRarget).beginSession();
+
+				setGlobalSearch(true);
+			}
+		}
+
+		initIncrementalBaseLocation();
+	}
+
+	public void endSession() {
+		if (target != null && target instanceof IFindReplaceTargetExtension)
+			((IFindReplaceTargetExtension) target).endSession();
+
+		target = null;
+	}
+
+
+	public void deactivateScope() {
+		if (target != null && (target instanceof IFindReplaceTargetExtension))
+			((IFindReplaceTargetExtension) target).setScope(null);
+
+		oldScope = null;
+	}
+
+	public String getCurrentSelection() {
+		if (target == null) {
+			return null;
+		}
+
+		return target.getSelectionText();
+	}
+
+	/**
+	 * Returns whether the target is editable.
+	 *
+	 * @return <code>true</code> if target is editable
+	 */
+	public boolean isEditable() {
+		boolean isEditable = (target == null ? false : target.isEditable());
+		return isTargetEditable && isEditable;
+	}
+
+	/**
+	 * @return Whether the target supports multiple selections
+	 */
+	public boolean supportsMultiSelection() {
+		return target instanceof IFindReplaceTargetExtension4;
+	}
+
+	/**
+	 * @return Whether the target is available
+	 */
+	public boolean isTargetAvailable() {
+		return target != null;
+	}
+
+	/**
+	 * Sets the given status message in the status line.
+	 *
+	 * @param error         <code>true</code> if it is an error
+	 * @param dialogMessage the message to display in the dialog's status line
+	 * @param editorMessage the message to display in the editor's status line
+	 */
+	private void statusMessage(boolean error, String dialogMessage, String editorMessage) {
+		status.setError(error);
+		status.setMessage(dialogMessage);
+
+		IEditorStatusLine statusLine = getStatusLineManager();
+		if (statusLine != null)
+			statusLine.setMessage(error, editorMessage, null);
+	}
+
+	/**
+	 * Sets the given error message in the status line.
+	 *
+	 * @param message the message
+	 */
+	private void statusError(String message) {
+		statusMessage(true, message, message);
+	}
+
+	/**
+	 * Sets the given message in the status line.
+	 *
+	 * @param message the message
+	 */
+	private void statusMessage(String message) {
+		statusMessage(false, message, message);
+	}
+
+	/**
+	 * Updates the search result after the Text was Modified. Used in combination
+	 * with <code>setIncrementalSearch(true)</code>
+	 *
+	 * @param searchString the String that is to be searched
+	 */
+	public void updateSearchResultAfterTextWasModified(String searchString) {
```

Comment:
What does the work "Text" in the method name mean? Is it about "TargetContent" or the like?